dotCMS/src/test/java/com/dotcms/content/elasticsearch/ESQueryCacheTest.java

```diff
@@ -0,0 +1,272 @@
+        
+        
+    }
+    
+    
+    
+    private SearchRequest getSearchRequest() {
+
+        return getSearchRquest(RandomStringUtils.randomAscii(200), true) ;
+        
+    }
+    
+    @NotNull
+    private SearchRequest getSearchRquest(String query, boolean live) {
+        SearchRequest searchRequest = new SearchRequest();
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(QueryBuilders.queryStringQuery(query));
+        searchSourceBuilder.size(0);
+        searchSourceBuilder.timeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS));
+        searchRequest.source(searchSourceBuilder);
+        searchRequest.indices((live ? "live" : "working"));
+        return searchRequest;
+    }
+
+    static DotCacheAdministrator admin = new DotCacheAdministrator() {
+
+        private Map<String, SearchHits> hitMap = new HashMap<>();
+
+        @Override
+        public void shutdown() {
+
+
+        }
+
+        @Override
+        public void setTransport(CacheTransport transport) {
+
+
+        }
+
+        @Override
+        public void removeLocalOnly(String key, String group, boolean ignoreDistributed) {
+            hitMap.remove(key + group);
+
+        }
+
+        @Override
+        public void remove(String key, String group) {
+            hitMap.remove(key + group);
+
+        }
+
+        @Override
+        public void put(String key, Object content, String group) {
+            hitMap.put(key + group, (SearchHits) content);
+
+        }
+
+        @Override
+        public void invalidateCacheMesageFromCluster(String message) {
+
+
+        }
+
+        @Override
+        public void initProviders() {
+
+
+        }
+
+        @Override
+        public CacheTransport getTransport() {
+
+            return null;
+        }
+
+        @Override
+        public DotCacheAdministrator getImplementationObject() {
+
+            return null;
+        }
+
+        @Override
+        public Class getImplementationClass() {
+
+            return null;
+        }
+
+        @Override
+        public Set<String> getGroups() {
+
+            return ImmutableSet.of(cache.getPrimaryGroup());
+        }
+
+        @Override
+        public List<CacheProviderStats> getCacheStatsList() {
+
+            return null;
+        }
+
+        @Override
+        public Object get(String key, String group) throws DotCacheException {
+
+            return hitMap.get(key + group);
+        }
+
+        @Override
+        public void flushGroupLocalOnly(String group, boolean ignoreDistributed) {
+            hitMap.clear();
+
+        }
+
+        @Override
+        public void flushGroup(String group) {
+
+            hitMap.clear();
+        }
+
+        @Override
+        public void flushAll() {
+            hitMap.clear();
+
+        }
+
+        @Override
+        public void flushAlLocalOnly(boolean ignoreDistributed) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'ignoreDistributed' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)